code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -377,8 +377,14 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 			payload.trace_gases = list()
 			payload.trace_gases += trace_gas
 
-			trace_gas.moles += min(total_oxygen/32,5)
-			total_oxygen -= min(trace_gas.moles*32,total_oxygen)
+			// 200/1024=0.19moles  0.19moles/12cells=0.0162moles per cell
+			// At 0.162 moles per cell it will take 21 iterations for reaction rate to drop below MINIMUM_REACT_QUANTITY
+			// Providing 1.4 minutes of catalyst assuming 4 sec ATMOS for a 12 cell burn chamber
+			trace_gas.moles += min(total_oxygen/1024,5)
+			total_oxygen -= min(trace_gas.moles*1024,total_oxygen)
```

Comment:
Won't this zero out `total_oxygen` after the first time this is procced?

Reply:
Yeah... updated to require 2 iterations.  Having oxygen be part of the merged gas will actually cause the temperature change to be meaningful.